### ISSUE TITLE
Support for allocation strategy attribute on AWS Batch compute environments

### ISSUE THREAD

**Issue by pgieser** [terraform-providers/terraform-provider-aws#10980]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Recently AWS Batch has added an allocation strategy configuration option for defining compute environments. Support for this new attribute would be helpful.

### New or Affected Resource(s)

* aws_batch_compute_environment

### Potential Terraform Configuration

```hcl
resource "aws_batch_compute_environment" "compute_environment" {
  compute_resources {
    allocation_strategy = "SPOT_CAPACITY_OPTIMIZED"
  }
}
```

### References

* https://docs.aws.amazon.com/batch/latest/userguide/allocation-strategies.html

**Comment by pgieser** [comment]
Looks like this is already being worked on - https://github.com/terraform-providers/terraform-provider-aws/issues/10894

**Comment by bflad** [comment]
Support for the new `compute_resources` configuration block `allocation_strategy` argument has been merged and will release with version 2.44.0 of the Terraform AWS Provider, Thursday next week. Thanks to @sam-burrell for the implementation. 👍
